sapp/ui/frontend/src/Filter.js

```diff
@@ -780,7 +784,65 @@ const SaveFilterModal = (
     return <Alert type="error">{error.toString()}</Alert>;
   }
 
-  const onOk = (): void => {
+  let filterList = [];
+
+  const importProps = {
+    multiple: true,
+    onRemove: filterFile => {
+      filterList.splice(filterList.indexOf(filterFile), 1);
+    },
+    beforeUpload: filterFile => {
+      if (filterFile.type !== 'application/json') {
+        message.error(`${filterFile.name} is not a json file`);
+      } else {
+        filterList = [...filterList, filterFile];
+      }
+      return filterFile.type === 'application/json' ?
+        false : Upload.LIST_IGNORE;
+    },
+    filterList,
+  };
+
+  const onImport = () => {
+    let count = filterList.length;
+    let skipped = 0;
+    filterList.forEach(filterFile => {
+      let reader = new FileReader();
+      reader.onload = (e) => {
+        let filter: FilterDescription = JSON.parse(e.target.result);
+        if (filter["name"] === undefined ||
+          filter["description"] === undefined ||
+          Object.keys(filter).length <= 2
+        ) {
+          message.error(`Invalid filter file, skipping import of ${filterFile.name}`);
+          skipped++;
+        } else {
+          let name = filter.name;
+          let description = filter.description;
+          delete filter["name"];
+          delete filter["description"];
+          saveFilter({
+            variables: {
+              name: name,
+              description: description,
+              json: JSON.stringify(filter),
+            },
+          });
+        }
+        count--;
+        if(count === 0) {
```

Comment:
Do we need to decrement and keep track of `count` here? Would we be able to have the following code block after the for each loop ends instead?
```javascript
          props.hideImportModal();
          let imported = filterList.length - skipped;
          if(imported !== 0) {
            message.success(`Imported ${imported} filters`);
          }
```

Reply:
Yes makes better sense! I will try this out.

Reply:
Done